lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -17,11 +17,8 @@ def run(&block)
 
           @workers = thread_count.times.map do
             Thread.new do
-              begin 
-                while item = queue.pop(true)
-                  yield item
-                end
-              rescue ThreadError
+              while !queue.empty? && item = queue.pop(true)
```

Comment:
The parens are a very good point. But there is no `#any?` defined on `Queue`[http://ruby-doc.org/core-2.2.3/Queue.html].